source/MRMesh/MRMeshDistance.cpp

```diff
@@ -263,6 +263,32 @@ MeshSignedDistanceResult findSignedDistance( const MeshPart & a, const MeshPart
     return (signedRes.signedDist > 0.0f) ? MeshSignedDistanceResult{res.a, res.b, 0.0f} : signedRes;
 }
 
+MRMESH_API float findMaxDistance( const Mesh& refMesh, const Mesh& mesh, const AffineXf3f* rigidB2A )
+{
+    MR_TIMER;
+
+    const auto& meshVerts = mesh.points;
+
+    std::vector<float> vertColorFloats( meshVerts.size() );
+
+    const auto objXf = rigidB2A ? *rigidB2A : AffineXf3f();
+
+    //call refmesh-point distance finder for each point
+    std::atomic<float> res( 0 );
+    BitSetParallelFor( mesh.topology.getValidVerts(), [&] ( VertId i )
+    {
+        const auto p = objXf( meshVerts[VertId( i )] );
+        if ( auto pp = refMesh.projectPoint( p ) )
+        {
+            float temp = sqrt( pp->distSq );
+            if ( temp > res )
+                res = temp;
+        }
+    });
```

Comment:
- better fiil `vertColorFloats` (and rename it to distances) with distances and then after parallel block find maximum in the vector
 - do not operate with sqrt, better work with distanceSq directly